src/tqec/plaquette/plaquette.py

```diff
@@ -105,6 +92,16 @@ def project_on_boundary(
             new_plaquette_qubits, new_scheduled_circuit, self.mergeable_instructions
         )
 
+    def __eq__(self, rhs: object) -> bool:
+        return (
+            isinstance(rhs, Plaquette)
+            and self.qubits == rhs.qubits
+            and self.circuit == rhs.circuit
+        )
+
+    def __hash__(self) -> int:
+        return hash((self.qubits, self.circuit))
```

Comment:
It is not, good catch! I wanted to take the hash of the string representation as a first good enough implementation. I will implement that directly in `ScheduledCircuit`.

Reply:
I am quite reluctant to implement `__hash__` on a mutable data-structure as this is calling for hard-to-find bugs later. Right now, because `Plaquette` is frozen, I will simply implement the hash using the string representation of the stored `ScheduledCircuit`. In the future, it might be interesting to provide a `FrozenScheduledCircuit` data-structure to ensure that it is actually never changed.

Reply:
See #358